### DISCUSSION TITLE
Initial Setup Failing

### DISCUSSION THREAD

**Discussion by EJansenMSFT** [jertel/vuegraf#47] [Q&A] [answered]
Hi Team,

Just found this project and thought i'd give it a go, but in trying to get up and running, I'm having an issue that I'm hoping someone can help me out with.  After reading through everything, I just customized the sample docker-compose.yaml and vuegraf.json files.  

After doing that, I just run:

docker-compose -f docker-compose.yaml build --pull && docker-compose -f docker-compose.yaml up -d

From there, it builds all of the containers, as seen here, except the vuegraf container just constantly restarts:

![image](https://user-images.githubusercontent.com/72317762/119936270-e3826b80-bf56-11eb-8521-34d86c7aa062.png)

It doesn't really give an error, it just says the following:

![image](https://user-images.githubusercontent.com/72317762/119936346-06ad1b00-bf57-11eb-8f7c-5e74aae4f071.png)

I just used the default "influxDB" portion within the .json file, as I've never used InfluxDB before, so i'm not sure if that has something to do with it or not?

![image](https://user-images.githubusercontent.com/72317762/119936461-3bb96d80-bf57-11eb-9c5e-a080ce008efa.png)

I found this in the vuegraf.py file:

![image](https://user-images.githubusercontent.com/72317762/119936687-8e932500-bf57-11eb-9732-36cd0d782c3d.png)

What am I doing wrong?

Thanks ,
Eric

**Comment by jertel** [comment, marked as answer]
Try updating the vuegraf.json file with `"host": "influxdb",` instead of my.influxdb.hostname.

**Reply by EJansenMSFT** [reply]
For some reason it's still doing the same thing:

![image](https://user-images.githubusercontent.com/72317762/120010816-97ade180-bfab-11eb-8d9e-d0facf45171e.png)

 :/

**Reply by jertel** [reply]
Is your grafana instance able to connect to the influxdb?

**Comment by EJansenMSFT** [comment]
Well, I can see that the influxdb is listening on port 8086 on the server here:

![image](https://user-images.githubusercontent.com/72317762/120017370-cfb92280-bfb3-11eb-976e-64818f89b8dc.png)

I've never used either Grafana or InfluxDB before so it's a bit of a learning curve.  I can see that if I log into the Graphana page though that i can add a data source.  In plugging the info in that was in the .json file, it appears to be good:

![image](https://user-images.githubusercontent.com/72317762/120018214-dac08280-bfb4-11eb-9066-3633fa876e22.png)

**Reply by jertel** [reply]
Ok that's encouraging. Perhaps try using that 192.168.10.20 as the host instead of 
"influxdb". Also, you can modify the docker-compose.yaml file to have Vuegraf startup with a generic "sleep 9999999" entrypoint. This would stop the crash loop and give you time to run `docker exec -it vuegraf /bin/sh` in another window. With that you have a terminal to see if the vuegraf container can access the influxdb, or run other diagnostics. This sounds like a simple problem but it's difficult to troubleshoot over a forum, but I'm sure we'll figure it out.

**Reply by EJansenMSFT** [reply]
Yeah, after doing the test in there with the server IP, I tried to modify the host with the IP and it didn't work.  Earlier I also tried to just hashtag this out as well:

![image](https://user-images.githubusercontent.com/72317762/120019260-38090380-bfb6-11eb-8dde-88601f21fda6.png)

I figured i'd just manually restart the container.  When I did that, it would just boot and shutdown.  So I'd start it, and then it'd shutdown, which made sense as to why we'd see the loop.  

Yeah, I'll keep playing with it - it doesn't help that my Unix / Linux knowledge is about zero and I also have no experience with graphana or influxdb.  I mention that because once I console in doing the exec -it, I wouldn't know what to do after that ;)

**Reply by jertel** [reply]
Perhaps you should try running Vuegraf on your host machine to see if this is a docker networking problem. If you git clone the vuegraf repo to your host machine and then run:
```
python3 src\vuegraf.py c:\path\to\your\vuegraf.json
```
I'm guessing on the path syntax since I don't normally run Python on Windows but it's probably something similar to that.

**Reply by jertel** [reply]
Good news - sort of. I spun up a Windows machine, installed docker and git/bash. I'm now able to reproduce the problem you're seeing. You're probably the first person that's attempted to do this on Windows, or at least the first person that brought it up here. I'll dig into it this weekend.

**Reply by EJansenMSFT** [reply]
Eww ok, that sounds like a win.  Yeah when I start playing around with a lot of these GitHub projects, I'm usually the only guy doing it on Windows and it throws the dev's off a bit ;)  I speak PowerShell, and they usually speak Python or Java.

**Reply by jertel** [reply]
I had a few minutes to look into it and I found that the `docker logs vuegraf` command on Windows doesn't actually show stderr from a crash. This makes things very difficult to debug. Once I trapped the underlying error I found that the cause was due to a bad username and password for Emporia on my Windows vuegraf.json configuration file. So I suggest you double check those Emporia credentials. Therefore, the InfluxDb path we were chasing is likely not the problem. 

I've pushed a change to ensure that all crashes log something to stdout, so that this isn't so difficult to troubleshoot. Make sure you first run `docker pull jertel/vuegraf` before re-running `docker-compose-run.sh` so that you can see this new error output.

**Reply by EJansenMSFT** [reply]
The username and password was actually something that I verified last night by logging out of the app and back in, before entering the data into the json file, so it _should_ be good.  I'll do the pull and then re-run the setup to see if that's what it's showing as the error output.

**Reply by EJansenMSFT** [reply]
Ah yes, waay better logging now:

![image](https://user-images.githubusercontent.com/72317762/120051250-e3817a80-bfed-11eb-9b39-cd8bdb8f0e56.png)

**Reply by jertel** [reply]
Ah. It's trying to hit `api.emporiaenergy.com` over https. Check your outbound firewall rules, or whatever else is blocking it. Should be good once you find that restriction.

**Reply by jertel** [reply]
Oops, I read that too quickly. It's still stuck on InfluxDB isn't it.. Try putting that back to `influxdb` and see if you get past that error.

**Reply by EJansenMSFT** [reply]
Sure enough:

![image](https://user-images.githubusercontent.com/72317762/120051592-3c9dde00-bfef-11eb-9b8e-ea434a5ccfec.png)

Hmmmmm..  I'll maybe try to re-type it in there.  It should be the same username / password that I use in the app, I would think.  I don't know if there's separate creds on their site or not.

**Reply by EJansenMSFT** [reply]
HAH, nope, it was me - one character had the case off.  That fixed it immediately:

![image](https://user-images.githubusercontent.com/72317762/120051755-e9785b00-bfef-11eb-97ca-b7a0c5e6523a.png)

Great support!  Thanks a ton!

Now I just need to figure everything else out. ;)
